infrap4d/daemon/daemon-unix.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2008, 2009, 2010, 2011, 2012, 2013, 2015 Nicira, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "daemon.h"
+#include <stdio.h>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <grp.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "fatal-signal.h"
+
+#ifdef __linux__
+#define LINUX 1
+#else
+#define LINUX 0
+#endif
+
+/* --detach: Should we run in the background? */
```

Comment:
It just dawned on me: we're linking the daemon code into infrap4d, along with stratum, krnlmon, and a p4 driver. We should ensure that we do not pollute the global namespace in the process.

One solution would be make these files C++, rather than C, and wrap them in namespaces. I'd suggest `inf4ap4d`.

Another solution would be to add a prefix to each global symbol. I would suggest `infrap4d_`.

We do not (necessarily) need to do this for the initial commit, but we should plan to follow up fairly soon.

Reply:
You are right, added infrap4d_ prefix to avoid any conflicts for global variables